server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -314,17 +317,17 @@ public String resolveDateMathExpression(String dateExpression) {
      * <p>Only aliases with filters are returned. If the indices list contains a non-filtering reference to
      * the index itself - null is returned. Returns {@code null} if no filtering is required.
      */
-    public String[] filteringAliases(ClusterState state, String index, String... expressions) {
-        return indexAliases(state, index, AliasMetaData::filteringRequired, false, expressions);
+    public Function<String, String[]> filteringAliases(ClusterState state, String... expressions) {
```

Comment:
Note to reviewers: please pay special attention to changes in this file where I reorganized loops to make things perform better.